resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala

```diff
@@ -749,13 +749,52 @@ private[spark] object Config extends Logging {
       .checkValue(value => value > 0, "Gracefully shutdown period must be a positive time value")
       .createWithDefaultString("20s")
 
+  val KUBERNETES_INGRESS_ENABLED =
+    ConfigBuilder("spark.kubernetes.driver.ingress.enabled")
+      .doc("Whether to create ingress entry of the driver service for external UI access " +
+        "in cluster mode. This only takes effect when `spark.ui.enabled` is true.")
+      .version("4.0.0")
+      .booleanConf
+      .createWithDefault(false)
+
+  val KUBERNETES_INGRESS_HOST_PATTERN =
+    ConfigBuilder("spark.kubernetes.driver.ingress.host")
+      .doc("Host for driver UI ingress, {{APP_ID}} will be replaced by application ID.")
+      .version("4.0.0")
+      .stringConf
+      .createOptional
+
+  val KUBERNETES_INGRESS_CLASS_NAME =
+    ConfigBuilder("spark.kubernetes.driver.ingress.ingressClassName")
+      .doc("Reference the IngressClass that should be used to implement the ingress for spark UI.")
+      .version("4.0.0")
+      .stringConf
+      .createOptional
+
+  val KUBERNETES_INGRESS_PATH =
+    ConfigBuilder("spark.kubernetes.driver.ingress.path")
```

Comment:
May I ask when you want to override this? And, is it safe in terms of the security?

Reply:
internally, we only use path '/' and pathType  `ImplementationSpecific`, let me hardcode it and make it configurable if someone has real request in the future